nixos/modules/services/logging/syslog-ng.nix

```diff
@@ -79,14 +79,14 @@ in
   config = lib.mkIf cfg.enable {
     systemd.services.syslog-ng = {
       description = "syslog-ng daemon";
-      preStart = "mkdir -p /{var,run}/syslog-ng";
       wantedBy = [ "multi-user.target" ];
       after = [ "multi-user.target" ]; # makes sure hostname etc is set
       serviceConfig = {
         Type = "notify";
         PIDFile = pidFile;
         StandardOutput = "null";
         Restart = "on-failure";
+        ExecStartPre = "${lib.getExe' pkgs.coreutils "mkdir"} -p /var/syslog-ng /run/syslog-ng";
```

Comment:
Can this use `systemd.tmpfiles` instead?

Reply:
I have some other PRs coming up for moving things back and forth from `systemd.tmpfiles`, `*Directory` and systemd hooks (#450232 is one). For now, I think I'll just move these without editing, and then leave them to be picked up later.

Reply:
Good call!